src/backend/columnar/columnar_customscan.c

```diff
@@ -182,21 +182,125 @@ ColumnarSetRelPathlistHook(PlannerInfo *root, RelOptInfo *rel, Index rti,
 		/* columnar doesn't support parallel paths */
 		rel->partial_pathlist = NIL;
 
+		/*
+		 * Remove index paths and recreate them by temporarily setting
+		 * enable_indexonlyscan to false to make sure that we don't use
+		 * index-only scan on columnar tables.
+		 *
+		 * Before creating index paths, create a sequential scan path as
+		 * set_plain_rel_pathlist does. This is because, it is not guaranteed
+		 * that postgres would generate an index path for regular index scan
+		 * and might already removed other scan path(s) when adding index-only
+		 * path(s). In that case, we should be able to safely fall-back to
+		 * sequential scan as postgres does.
+		 */
+		RemovePathsByPredicate(rel, IsIndexPath);
+		CreateSyncSeqScanPath(root, rel);
+		CreateNonIndexOnlyIndexPaths(root, rel);
```

Comment:
Here I initially thought that we should just drop & recreate the index paths.
Then I realized that adding a path `(add_path)` might remove the other paths in the `rel->pathlist`.
https://github.com/postgres/postgres/blob/6ee41a301e70fc8e4ad383bad22d695f66ccb0ac/src/backend/optimizer/util/pathnode.c#L582..L584
Then, following what `set_plain_rel_pathlist` does, I decided to create seq paths too, in case seq scan paths are dominated by some other but removed by the calls made to `add_path`.

On the other hand, following `set_plain_rel_pathlist` doesn't seem to be enough. This is because, there is another function that postgres calls to create the paths via `set_rel_pathlist`, that is `set_append_rel_pathlist`.
For example, for partitioned tables, postgres calls `set_append_rel_pathlist` to create the paths and (maybe index) paths created by `set_append_rel_pathlist` will not be re-created by us.

Given that `set_append_rel_pathlist` is a static postgres function, I'm not sure how to proceed from here.

Reply:
maybe directly set `allvisfrac` to 0